### PR TITLE
HR build fixes

### PR DESCRIPTION
**Pull request by haiodo**
Signed-off-by: Andrey Sobolev <haiodo@gmail.com>

### CODE REVIEW

File: dev/server/src/index.ts [hunk 1]
```diff
@@ -32,7 +33,7 @@ async function start (): Promise<void> {
 
   const db = client.db('accounts')
 
-  await upgradeWorkspace(defaultWorkspace, { mongoDBUri: dbUri })
+  await upgradeWorkspace(defaultWorkspace, { mongoDBUri: dbUri, txes: builder.getTxes() })
```
Comment (keepitsimple) [inline dev/server/src/index.ts:36]: It gives me error:
 TS2345: Argument of type '{ mongoDBUri: string; txes: Tx<Doc>[]; }' is not assignable to parameter of type 'WorkspaceOptions'.   Object literal may only specify known properties, and 'txes' does not exist in type 'WorkspaceOptions'.

File: plugins/chunter-impl/src/components/CreateDirectMessage.svelte [hunk 2]
```diff
@@ -44,32 +44,35 @@
...
-        const m2 = [client.accountId(), ...to].sort()
-        if (deepEqual(m1, m2)) {
-          // Ok, we have channel already
-          currentSpace = c
-          break
+  // Check if we have already a channel between us.
+  $: client.findAll<Space>(chunter.class.Channel, { direct: true, private: true }).then((channels) => {
+    for (const c of channels) {
+      const m1 = [...c.members].sort()
+      const m2 = [client.accountId(), ...to].sort()
```
Comment (keepitsimple) [inline plugins/chunter-impl/src/components/CreateDirectMessage.svelte:63]: does it makes sense to sort it 1 time before the for loop?

File: plugins/chunter-impl/src/components/UserSpaceInfo.svelte [hunk 3]
```diff
@@ -55,7 +55,7 @@
     <div class="caption">
       <div class="title" class:header={headerSize} class:item={!headerSize}>
         {accounts
-          .map((u) => u.name)
+          .map((u) => (u.name.trim().length > 0 ? u.name : u.email))
```
Comment (keepitsimple) [inline plugins/chunter-impl/src/components/UserSpaceInfo.svelte:58]: `.map((u) => (u.name.trim().length > 0 ? u.name.trim() : u.email))` ?

File: products/hr-build/src/server.ts [hunk 4]
```diff
@@ -13,7 +17,36 @@ const appSecret = process.env.SERVER_SECRET ?? 'secret'
 
 const dbUri = process.env.MONGODB_URI ?? 'mongodb://localhost:27017'
 
+const WORKSPACE = process.env.WORKSPACE ?? 'workspace'
+const WORKSPACE_ORGANIZATION = process.env.WORKSPACE_ORGANIZATION ?? ''
```
Comment (keepitsimple) [inline products/hr-build/src/server.ts:21]: `const WORKSPACE_ORGANIZATION = process.env.WORKSPACE_ORGANIZATION ?? 'Organization '` ?

File: dev/server/src/index.ts (hunk 1 above)
Comment (haiodo) [reply dev/server/src/index.ts:36]: Please do `rush build`

File: plugins/chunter-impl/src/components/CreateDirectMessage.svelte (hunk 2 above)
Comment (haiodo) [reply plugins/chunter-impl/src/components/CreateDirectMessage.svelte:63]: Thanks.

File: plugins/chunter-impl/src/components/UserSpaceInfo.svelte (hunk 3 above)
Comment (haiodo) [reply plugins/chunter-impl/src/components/UserSpaceInfo.svelte:58]: If name is not specified, email will be used.

File: products/hr-build/src/server.ts (hunk 4 above)
Comment (haiodo) [reply products/hr-build/src/server.ts:21]: Not required I think.

File: server/server/package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@anticrm/server",
   "version": "0.4.0",
-  "main": "lib/index.js",
+  "main": "src/index.ts",
```
Comment (BykhovDenis) [inline server/server/package.json:4]: Why?

**Review by BykhovDenis — APPROVED** [review]

File: server/server/package.json (hunk 5 above)
Comment (haiodo) [reply server/server/package.json:4]: ￼￼￼
Server is not used by its own, so it doesn't really matter for now, it just faster to develop server and start dev/server

File: plugins/chunter-impl/src/components/UserSpaceInfo.svelte (hunk 3 above)
Comment (keepitsimple) [reply plugins/chunter-impl/src/components/UserSpaceInfo.svelte:58]: @haiodo No. please look at `? u.name.trim() : `

File: products/hr-build/src/server.ts (hunk 4 above)
Comment (keepitsimple) [reply products/hr-build/src/server.ts:21]: ok

**Review by sumboid — APPROVED** [review]

File: plugins/chunter-impl/src/components/CreateDirectMessage.svelte [hunk 6]
```diff
@@ -44,32 +44,35 @@
   let accountUpdater: QueryUpdater<Account> | undefined = undefined
 
   $: accountUpdater = client.query<Account>(accountUpdater, core.class.Account, {}, (results) => {
-    allAccounts = results
+    allAccounts = results.filter((it) => it._id !== client.accountId())
```
[the remark this answers is not in the document]
Comment (haiodo) [reply plugins/chunter-impl/src/components/CreateDirectMessage.svelte:47]: Nope, created issue: https://github.com/hardcoreeng/platform/issues/342

**Pull request merged**